### PR TITLE
Add r-liger

### PR DESCRIPTION
**Pull request by raivivek**
Signed-off-by: Vivek Rai <mail@raivivek.in>

**Replace this sentence with a detailed description of your pull request**

----

> Bioconda requires reviews prior to merging pull-requests (PRs). To facilitate this, once your PR is passing tests and ready to be merged, please add the `please review & merge` label so other members of the bioconda community can have a look at your PR and either make suggestions or merge it.

* [x] I have read the [guidelines for bioconda recipes](https://bioconda.github.io/contributor/guidelines.html).
* [x] This PR adds a new recipe.
* [x] AFAIK, this recipe **is directly relevant to the biological sciences**
      (otherwise, please submit to the more general purpose [conda-forge channel](https://conda-forge.org/docs/)).
* [ ] This PR updates an existing recipe.
* [ ] This PR does something else (explain below).

> Everyone has access to the following BiocondaBot commands, which can be given in a comment:
>
>  * `@BiocondaBot please update` will cause the BiocondaBot to merge the master branch into a PR
>  * `@BiocondaBot please add label` will add the `please review & merge` label.
>  * `@BiocondaBot please fetch artifacts` will post links to packages and docker containers built by the CI system. You can use this to test packages locally before merging.
>
> For members of the Bioconda project, the following command is also available:
>
>  * `@BiocondaBot please merge` will cause packages/containers to be uploaded and a PR merged. Someone must approve a PR first! This has the benefit of not wasting CI build time required by manually merging PRs.
>
> If you have questions, please post them in gitter or ping `@bioconda/core` in a comment (if you are not able to directly ping `@bioconda/core` then the bot will repost your comment and enable pinging).

### CODE REVIEW

**Comment by raivivek** [comment]
@bioconda/build-system Any tips on why the build gives an error that `cannot find CMD`?
```
19:44:55 BIOCONDA INFO (OUT) /opt/conda/conda-bld/r-liger_1580845382655/work/conda_build.sh: line 4: CMD: command not found
```

**Comment by mbargull** [comment]
>  why the build gives an error that `cannot find CMD`

`r-base` was not in the `build`/`host` environment hence `conda-build` did not set the `R` environment variable thus the command `$R CMD` became just ` CMD`.

**Comment by raivivek** [comment]
Arrgh! Thanks for pointing that out and fixing it. @mbargull

**Comment by raivivek** [comment]
Hi @mbargull, looks like one of the dependencies cannot be found (r-rann.l1) and hence the build fails. I see that this package is available with the r-channel, but I am assuming that's not pulled in. What would you suggest to fix this? Thank you.

**Comment by raivivek** [comment]
Okay, looks like I was able to get it to work. Not sure if that's the best way. @bioconda-bot please add label

File: recipes/r-liger/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+test:
+  commands:
+    - $R -e "library('liger')"           # [not win]
+    - "\"%R%\" -e \"library('liger')\""  # [win]
+
+about:
+  home: https://github.com/MacoskoLab/liger
+  license: GPL-3
+  summary: |
+    Uses an extension of nonnegative matrix factorization to identify shared and
```
Comment (dpryan79) [inline recipes/r-liger/meta.yaml]: Can you explicitly mention that this is relevant for single-cell data in the summary?

File: recipes/r-liger/build.sh [hunk 2]
```diff
@@ -0,0 +1,2 @@
+wget 'https://cran.r-project.org/src/contrib/Archive/RANN.L1/RANN.L1_2.5.2.tar.gz' && $R CMD INSTALL RANN.L1_2.5.2.tar.gz
```
[the remark this answers is not in the document]
Comment (raivivek) [reply recipes/r-liger/build.sh]: @dpryan79 Okay, I have added a PR here: https://github.com/conda-forge/staged-recipes/pull/10772

**Comment by raivivek** [comment]
@dpryan79 I have an unrelated question for a different package (qorts) -- how do we decided which Python version the package gets built for? For instance, the qorts package has no reason to not work for Python 3.7 version but it doesn't install through Conda for Python 3.7.

**Comment by dpryan79** [comment]
@bioconda-bot please fetch artifacts

**Comment by dpryan79** [comment]
@bioconda-bot please merge

**Comment by raivivek** [comment]
Thanks @dpryan79!

**Pull request merged**